### PR TITLE
MQTT blog post

### PR DESCRIPTION
**Pull request by fabaff**
This blog post shows three ways to use MQTT with HA.

### CODE REVIEW

**Comment by fabaff** [comment]
It would be nice to get some feedback. Thanks.

File: source/_posts/2015-09-10-mqtt-different-ways-to-use-mqtt-with-home-assistant.markdown [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+description: "This post describes three different ways to use MQTT with Home Assistant."
+date: 2015-09-10 11:19:38 +0200
+date_formatted: "September 10, 2015"
+author: Fabian Affolter
+comments: true
+categories: 
+og_image: /images/blog/2015-09-mqtt/arduino.png
+---
+
+The [MQTT](https://en.wikipedia.org/wiki/MQTT) support was added to Home Assistant recently. The [MQTT component](https://home-assistant.io/components/mqtt.html) will enable you to do all sort of things. Most likely you will use it to communicate with your devices. But Home Assistant don't care where the data is coming from or is limited to real hardware as long as there is MQTT support. This means that it doesn't matter if the data is coming from a human, a web service, or a device.
```
Comment (rmkraus) [inline source/_posts/2015-09-10-mqtt-different-ways-to-use-mqtt-with-home-assistant.markdown]: But home assistant *doesn't*

File: source/_posts/2015-09-10-mqtt-different-ways-to-use-mqtt-with-home-assistant.markdown [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+```yaml
+  - platform: mqtt
+    name: "Lottery"
+    state_topic: "home-assistant/lottery/number"
+    unit_of_measurement: " "
+```
+
+Don't forget to restart Home Assistant to make the configuration active.
+
+To play, we need numbers from 1 to 49 which can be marked on the ticket. Those number should be random and displayed in the Home Assistant frontend. The Python script below is another simple example on how to send MQTT messages from the commandline. This time in a loop. For further information and examples please check the [Paho MQTT](https://eclipse.org/paho/clients/python/docs/) documentation.
```
Comment (rmkraus) [inline source/_posts/2015-09-10-mqtt-different-ways-to-use-mqtt-with-home-assistant.markdown]: Those *numbers*

Comment (rmkraus) [inline source/_posts/2015-09-10-mqtt-different-ways-to-use-mqtt-with-home-assistant.markdown]: ; this time in a loop.

File: source/_posts/2015-09-10-mqtt-different-ways-to-use-mqtt-with-home-assistant.markdown [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+## {% linkable_title Arduino %}
+
+To get started with real hardware that is capable to send MQTT messages, the Arduino platform is an inexpensive way to do it. In this section an Arduino UNO with an Ethernet shield and a photo resistor is used. The photo resistor is connected to analog pin 0 (A0) and has an output from 0 to 1024.
+
+<p class='img'>
+  <img src='/images/blog/2015-09-mqtt/arduino-shield.png' />
+  The Arduino UNO with Ethernet shield and photo resistor
+</p>
+
+The [MQTT client](http://knolleary.github.io/pubsubclient/) for the Arduino needs to be available in your Arduino IDE. Below you find an sketch which could act as an starting point. Please modify the IP addresses, the MAC address, and the pin as needed and upload the sketch to your Arduino.
```
Comment (rmkraus) [inline source/_posts/2015-09-10-mqtt-different-ways-to-use-mqtt-with-home-assistant.markdown]: Below *you'll* find

*a* sketch

*a* starting point

File: source/_posts/2015-09-10-mqtt-different-ways-to-use-mqtt-with-home-assistant.markdown [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+```
+
+Every 5 seconds a message with a new number is sent to the broker and picked up by Home Assistant. By the way, my mood is much better now.
+
+<p class='img'>
+  <img src='/images/blog/2015-09-mqtt/lottery.png' />
+  The Lottery sensor
+</p>
+
+With only a few lines of Python and a MQTT broker you can create your own "smartdevice" or send information to Home Assistant which you haven't think of. Of course this is not limited to Python. If there is a MQTT library available, the device can be used with Home Assistant now.
```
Comment (rmkraus) [inline source/_posts/2015-09-10-mqtt-different-ways-to-use-mqtt-with-home-assistant.markdown]: *an* MQTT

File: source/_posts/2015-09-10-mqtt-different-ways-to-use-mqtt-with-home-assistant.markdown [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+#if DEBUG
+  Serial.print("Sensor value: ");
+  Serial.println(sensorValue);
+#endif
+  client.publish(stateTopic, itoa(sensorValue, buf, 10));
+  delay(updateInterval);
+}
+```
+
+The Arduino will sent the value of the sensor every second. To use the data in Home Assistant, add an additional MQTT sensor to the `configuration.yaml` file.
```
Comment (rmkraus) [inline source/_posts/2015-09-10-mqtt-different-ways-to-use-mqtt-with-home-assistant.markdown]: Will *send*

**Comment by rmkraus** [comment]
Very informative. Thanks.

File: source/_posts/2015-09-10-mqtt-different-ways-to-use-mqtt-with-home-assistant.markdown [hunk 6]
```diff
@@ -0,0 +1,201 @@
+---
+layout: post
+title: "MQTT: Different ways to use MQTT with Home Assistant"
+description: "This post describes three different ways to use MQTT with Home Assistant."
+date: 2015-09-10 11:19:38 +0200
+date_formatted: "September 10, 2015"
+author: Fabian Affolter
+comments: true
+categories: 
+og_image: /images/blog/2015-09-mqtt/arduino.png
```
Comment (balloob) [inline source/_posts/2015-09-10-mqtt-different-ways-to-use-mqtt-with-home-assistant.markdown]: This image should be included in the PR.

File: source/_posts/2015-09-10-mqtt-different-ways-to-use-mqtt-with-home-assistant.markdown [hunk 7]
```diff
@@ -0,0 +1,201 @@
+---
+layout: post
+title: "MQTT: Different ways to use MQTT with Home Assistant"
```
Comment (balloob) [inline source/_posts/2015-09-10-mqtt-different-ways-to-use-mqtt-with-home-assistant.markdown]: Let's remove the `MQTT:` at the start. I always aim to keep the titles on 1 line on the [blog page](https://home-assistant.io/blog/)

File: source/_posts/2015-09-10-mqtt-different-ways-to-use-mqtt-with-home-assistant.markdown [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+After a restart of Home Assistant the "Mood" sensor will show up in the frontend. For more details about the configuration of MQTT itself and the sensor, please refer to the [MQTT component](https://home-assistant.io/components/mqtt.html) or the [MQTT sensor](https://home-assistant.io/components/sensor.mqtt.html) documentation. 
+
+Now we can set the mood. The commandline tool (`mosquitto_pub`) which is shipped with `mosquitto` is used to send a MQTT message.
+
+```bash
+mosquitto_pub  -h 127.0.0.1 -t "home-assistant/fabian/mood" -m "bad"
+```
+
+<p class='img'>
+  <img src='/images/blog/2015-09-mqtt/mood.png' />
```
Comment (balloob) [inline source/_posts/2015-09-10-mqtt-different-ways-to-use-mqtt-with-home-assistant.markdown]: This image should be included with this PR.

File: source/_posts/2015-09-10-mqtt-different-ways-to-use-mqtt-with-home-assistant.markdown [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+og_image: /images/blog/2015-09-mqtt/arduino.png
+---
+
+The [MQTT](https://en.wikipedia.org/wiki/MQTT) support was added to Home Assistant recently. The [MQTT component](https://home-assistant.io/components/mqtt.html) will enable you to do all sort of things. Most likely you will use it to communicate with your devices. But Home Assistant don't care where the data is coming from or is limited to real hardware as long as there is MQTT support. This means that it doesn't matter if the data is coming from a human, a web service, or a device. 
+
+A great example is shown in a [Laundry Automation](https://home-assistant.io/blog/2015/08/26/laundry-automation-with-moteino-mqtt-and-home-assistant/) post in this blog. 
+
+This post will give you a small overview of some other possibilities on how to use MQTT with Home Assistant. 
+
+<!--more-->
```
Comment (balloob) [inline source/_posts/2015-09-10-mqtt-different-ways-to-use-mqtt-with-home-assistant.markdown]: Let's make sure we have an image above the read more button, makes the post more appealing. Want me to make a similar graphic like [the one for IFTTT](https://github.com/balloob/home-assistant.io/issues/45)?

File: source/_posts/2015-09-10-mqtt-different-ways-to-use-mqtt-with-home-assistant.markdown (hunk 6 above)
Comment (fabaff) [inline source/_posts/2015-09-10-mqtt-different-ways-to-use-mqtt-with-home-assistant.markdown]: Forgot to commit.

File: source/_posts/2015-09-10-mqtt-different-ways-to-use-mqtt-with-home-assistant.markdown (hunk 7 above)
Comment (fabaff) [inline source/_posts/2015-09-10-mqtt-different-ways-to-use-mqtt-with-home-assistant.markdown]: Changed.

File: source/_posts/2015-09-10-mqtt-different-ways-to-use-mqtt-with-home-assistant.markdown (hunk 9 above)
Comment (fabaff) [inline source/_posts/2015-09-10-mqtt-different-ways-to-use-mqtt-with-home-assistant.markdown]: A nice graphic would be very nice. Adding a sceenshot of a badge wouldn't be very appealing.

**Comment by fabaff** [comment]
Thank you, guys.

**Pull request merged**